File: restaurant-app/src/app/model/order/online-order-info.ts
import {Address} from "../address";
import {PaymentMethod} from "../payment-method";
import {OrderInfo} from "./order-info";
import {RestaurantShortInfo} from "../restaurant/restaurant-short-info";

export interface OnlineOrderInfo {
  id: number;
  restaurantInfo: RestaurantShortInfo
  name: string;
  surname: string;
  email: string;
  phoneNr: string;
  address: Address;
  floor: number | any;
  orderDate: Date;
  paid: boolean;
  deliveryDate: Date | any;
  paymentMethod: PaymentMethod;
  orders: OrderInfo[];
  price: number;
}

File: restaurantservice/order-service/src/main/java/pl/restaurant/orderservice/api/response/OnlineOrderInfo.java
package pl.restaurant.orderservice.api.response;

import lombok.*;
import pl.restaurant.orderservice.api.request.Address;
import pl.restaurant.orderservice.data.entity.PaymentMethod;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class OnlineOrderInfo {
    private Long id;
    private RestaurantShortInfo restaurantInfo;
    private String name;
    private String surname;
    private String email;
    private String phoneNr;
    private Address address;
    private Integer floor;
    private LocalDateTime orderDate;
    private boolean isPaid;
    private LocalDateTime deliveryDate;
    private PaymentMethod paymentMethod;
    private List<OrderInfo> orders;
    private BigDecimal price;
}
